src/components/Preferences/ScheduleSelector.tsx

```diff
@@ -2,7 +2,7 @@ import React, { useState } from "react";
 import ScheduleSelector from "react-schedule-selector";
 
 const Timetable = (props) => {
-    const { semester, values, setFieldValue } = props;
+    const { semester, values, setFieldValue, isDisabled } = props;
```

Comment:
make `false` default value for `isDisabled`, then don't need to always pass it in for enabled cases
```suggestion
    const { semester, values, setFieldValue, isDisabled = false } = props;
```